src/express.re

```diff
@@ -268,7 +268,10 @@ module App = {
   external asMiddleware : t => Middleware.t = "%identity";
   /** [asMiddleware app] casts an App instance to a Middleware type */
 
-  external listen : t => port::int => unit = "" [@@bs.send];
+  external listen_ : t => port::int => onListen::(Js.Null_undefined.t Js.Exn.t => unit) [@bs.uncurry] => unit = "" [@@bs.send];
+
+  let listen app ::port=3000 ::onListen=(fun _ => ()) => listen_ app ::port ::onListen;
```

Comment:
So there's no way to ship this without a breaking change, so we might as well make a nice one 😄 

Do you mind changing this and the listen_ function to be something like the following?

```reason
external listen_ : App.t => int => (Js.Null_undefined.t Js.Exn.t => unit) [@bs.uncurry] => unit =
  "listen" [@@bs.send];

let listen app ::port=3000 ::onListen=(fun _ => ()) () => listen_ app port onListen;
```